functions/player/fn_assembleReplayData.sqf

```diff
@@ -0,0 +1,76 @@
+#include "script_component.hpp"
+
+params [["_part",0],["_startIndex",0],["_startTime",diag_tickTime],["_currentUnitsDataStates",[]]];
+
+if (_part == 0) then {
+	INFO("Assembling replay data.");
+} else {
+	INFO_2("Continuing assembly at index %1 (recursion %2).",_startIndex,_part);
+};
+
+if (isNil "GRAD_REPLAY_DATABASE_LOCAL_ASSEMBLED") then {
+	GRAD_REPLAY_DATABASE_LOCAL_ASSEMBLED = [];
+};
+
+private _typeDefaults = [
+	"",		// icon
+	-1,		// color ID
+	[0,0],	// pos2D
+	-1,		// dir
+	"",		// name
+	""		// group name
+];
+
+private _interrupt = false;
+private _startTimePart = diag_tickTime;
+private _continueAt = 0;
+
+for [{_i=_startIndex},{_i< count GRAD_REPLAY_DATABASE_LOCAL},{_i=_i+1}] do {
+
+	_compressedIntervalData = GRAD_REPLAY_DATABASE_LOCAL select _i;
+    _intervalData = [];
+
+	{
+		// catch nil entries, not sure what's causing them
+		if (!isNil "_x") then {
+
+			// timestamp
+	        if (_x isEqualType 0) exitWith {
```

Comment:
indentation relative to the isNil check above.

Reply:
Also doesn't exitWith exit the loop?

Reply:
No, only exits the `if`.

Reply:
Ah yeah.. Overlooked that because of the indentation :D